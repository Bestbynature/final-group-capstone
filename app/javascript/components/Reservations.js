import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { setReservedFlights } from '../redux/flights/flightsSlice';
import AddFlight from './AddFlight';

const Reservations = () => {
  const dispatch = useDispatch();
  const { reservedFlights } = useSelector((store) => store.flights);

  useEffect(() => {
    if (reservedFlights.length === 0) {
      const data = JSON.parse(localStorage.getItem('reservedFlights'));
      if (data) data.map((flight) => dispatch(setReservedFlights(flight)));
    }
  }, [reservedFlights, dispatch]);

  return (
    <div className="reservations">
      <AddFlight />
      <h2>Reserved Destinations</h2>
      <hr />
      <div className="reserved-flights">
        {reservedFlights.map((reservation) => (
          <div className="reserved-flight" key={reservation.id}>
            <table>
              <tr>
                <td>
                  <h3>Name:</h3>
                  {' '}
                </td>
                <td><h3>{reservation.name}</h3></td>
              </tr>
              <tr>
                <td><h3>From:</h3></td>
                <td><h3>{reservation.city}</h3></td>
              </tr>
              <tr>
                <td><h3>To:</h3></td>
                <td><h3>{reservation.flight}</h3></td>
              </tr>
              <tr>
                <td><h3>Date of Departure:</h3></td>
                <td><h3>{reservation.date}</h3></td>
              </tr>
            </table>
          </div>
        ))}
      </div>

    </div>
  );
};

export default Reservations;
