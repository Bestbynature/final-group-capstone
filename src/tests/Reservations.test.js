import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // Import configureStore
import Reservations from '../../app/javascript/components/Reservations'; // Update the path

const mockStore = configureStore([]);

describe('Reservations Component', () => {
  test('renders the component with reserved flights', () => {
    const initialState = {
      flights: {
        reservedFlights: [
          // Your reserved flights data here
        ],
      },
    };

    const store = mockStore(initialState);

    // Render the component using MemoryRouter and Provider
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Reservations />
        </Provider>
      </MemoryRouter>,
    );

    // Assertions
    const reservedFlightElements = container.querySelectorAll('.reserved-flight');
    expect(reservedFlightElements.length).toBe(initialState.flights.reservedFlights.length);

    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});
