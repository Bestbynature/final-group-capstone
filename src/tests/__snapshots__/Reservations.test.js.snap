// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservations Component renders the component with reserved flights 1`] = `
<div>
  <div
    class="reservations"
  >
    <div
      aria-hidden="true"
      aria-labelledby="addModalLabel"
      class="modal fade"
      id="addModal"
      tabindex="-1"
    >
      <form>
        <div
          class="modal-dialog"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h1
                class="modal-title"
                id="addModalLabel"
              >
                Add a New Flight
              </h1>
              <button
                aria-label="Close"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              class="modal-body"
            >
              <input
                class="form-control mt-3"
                data-testid="flight-name-input"
                placeholder="Flight Name"
                required=""
                type="text"
                value=""
              />
              <input
                class="form-control mt-3"
                data-testid="flight-picture-input"
                placeholder="Picture URL for the flight"
                required=""
                type="text"
                value=""
              />
              <input
                class="form-control mt-3"
                data-testid="flight-base-price-input"
                placeholder="Base Price for the destination"
                required=""
                type="number"
                value=""
              />
              <input
                class="form-control mt-3"
                data-testid="flight-available-slots-input"
                placeholder="Available Slots for the destination"
                required=""
                type="number"
                value=""
              />
            </div>
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-danger form-control mt-2"
                data-bs-dismiss="modal"
                type="button"
              >
                Cancel
              </button>
              <button
                class="btn add-submit form-control mt-3"
                data-bs-dismiss="modal"
                data-testid="add-flight-form"
                type="submit"
              >
                Create a Flight
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <h2>
      Reserved Destinations
    </h2>
    <hr />
    <div
      class="reserved-flights"
    />
  </div>
</div>
`;
